src/components/CitationExporter/components/CustomFormatSelect.tsx

```diff
@@ -1,10 +1,93 @@
-import { Dispatch } from 'react';
+import { Dispatch, useEffect, useState } from 'react';
 import { CitationExporterEvent } from '../CitationExporter.machine';
+import { useStore } from '@store';
+import { Button, FormControl, FormLabel, Input } from '@chakra-ui/react';
+import { Select } from '@components/Select';
+import { ChangeEvent } from 'react';
 
 export interface ICustomFormatSelectProps {
   dispatch: Dispatch<CitationExporterEvent>;
 }
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-export const CustomFormatSelect = (props: ICustomFormatSelectProps) => {
-  return <div>Coming soon!</div>;
+export const CustomFormatSelect = ({ dispatch }: ICustomFormatSelectProps) => {
+  const customFormats = useStore((store) => store.settings.user?.customFormats ?? []);
+
+  // custom formats to options
+  const customFormatOptions = customFormats
+    .map((f) => ({
+      id: f.id,
+      label: f.name,
+      value: f.id,
+      code: f.code,
+    }))
+    .sort((a, b) => (a.label < b.label ? -1 : 1));
+  customFormatOptions.splice(0, 0, { id: 'new', label: 'Enter New Format', value: 'new', code: '%1H:%Y:%q' });
```

Comment:
This "enter new format" makes me think that it's going to save it somewhere for me.  maybe "Enter custom format" or something like that.  Unless the plan is to allow users to save a format from here

Reply:
Updated.